apps/system/js/launcher.js

```diff
@@ -0,0 +1,61 @@
+/* global BaseModule */
+'use strict';
+
+(function() {
+  var Launcher = function() {};
+  Launcher.SETTINGS = [
+    'ftu.manifestURL',
+    'homescreen.manifestURL',
+    'deviceinfo.previous_os',
+    'deviceinfo.os',
+    'lockscreen.enabled'
+  ];
+  Launcher.STATES = [
+    'isUpgrade'
+  ];
+  BaseModule.create(Launcher, {
+    name: 'Launcher',
+    ready: function() {
+      this._version = {};
+      return Promise.all([
+        this.readFtu(),
+        this.readSetting('homescreen.manifestURL'),
+        this.readSetting('ftu.manifestURL'),
+        this.readSetting('deviceinfo.os'),
+        this.readSetting('deviceinfo.previous_os')
+      ]).then(function(results) {
```

Comment:
just a suggestion, but maybe you can use destructing assignment to have more meaningful variable names instead of results[0,1,2,3,4]